pulsar-client-cpp/lib/MessageId.cc

```diff
@@ -92,8 +101,19 @@ int32_t MessageId::batchIndex() const { return impl_->batchIndex_; }
 int32_t MessageId::partition() const { return impl_->partition_; }
 
 PULSAR_PUBLIC std::ostream& operator<<(std::ostream& s, const pulsar::MessageId& messageId) {
-    s << '(' << messageId.impl_->ledgerId_ << ',' << messageId.impl_->entryId_ << ','
-      << messageId.impl_->partition_ << ',' << messageId.impl_->batchIndex_ << ')';
+    std::function<void(std::ostream&, const MessageIdImpl&)> printMsgIdImpl = [](std::ostream& s,
```

Comment:
Use `auto` instead of the long `function` definition. And we can add the `s` in the lambda capture list.

```c++
    auto printMsgIdImpl = [&s](const MessageIdImpl& impl) {
        s << '(' << impl.ledgerId_ << ',' << impl.entryId_ << ',' << impl.partition_ << ','
          << impl.batchIndex_ << ')';
    };
```